.github/workflows/update_release_notes.yml

```diff
@@ -0,0 +1,68 @@
+name: Update release notes
+
+on:
+  push:
+    branches:
+      - 'release-notes'
+
+jobs:
+  fetch_version_name_release_notes:
+    runs-on: ubuntu-latest
+    outputs:
+      version_name_release_notes: ${{ steps.fetch_version_name_release_notes.outputs.version_name_release_notes }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Fetch version name for updating release notes
+        id: fetch_version_name_release_notes
+        env:
+          PROJECT_ROOT: ${{ github.workspace }}
+        run: |
+          chmod +x scripts/fetch_release_notes_version_name.sh
+          VERSION_NAME=$(./scripts/fetch_release_notes_version_name.sh $PROJECT_ROOT)
+          echo -e "version_name_release_notes=$VERSION_NAME" >> $GITHUB_OUTPUT
+          echo -e $VERSION_NAME
+
+  fetch_updated_release_notes_file:
+    runs-on: ubuntu-latest
+    outputs:
+      updated_release_notes_filepath: ${{ steps.fetch_updated_release_notes_file.outputs.updated_release_notes_filepath }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Fetch filepath for updated release notes file
+        id: fetch_updated_release_notes_file
+        env:
+          PROJECT_ROOT: ${{ github.workspace }}
+        run: |
+          chmod +x scripts/fetch_updated_release_notes_file.sh
+          UPDATED_RELEASE_NOTES_FILEPATH=$(./scripts/fetch_updated_release_notes_file.sh $PROJECT_ROOT)
+          echo -e "updated_release_notes_filepath=$UPDATED_RELEASE_NOTES_FILEPATH" >> $GITHUB_OUTPUT
+          echo -e $UPDATED_RELEASE_NOTES_FILEPATH
+
+  update_release_notes:
+    runs-on: ubuntu-latest
+    needs: [fetch_version_name_release_notes, fetch_updated_release_notes_file]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Print updated release notes
+        run: |
+          cat ${{ needs.fetch_updated_release_notes_file.outputs.updated_release_notes_filepath }}
+
+      # Update Github release with the updated release notes
+      - name: Update Github release notes
+        id: update_github_release_notes
+        uses: ncipollo/release-action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          VERSION_NAME: ${{ needs.fetch_version_name_release_notes.outputs.version_name_release_notes }}
+        with:
+          token: ${{ env.GITHUB_TOKEN }}
+          tag: ${{ env.VERSION_NAME }}
+          allowUpdates: true
+          updateOnlyUnreleased: true
+          omitNameDuringUpdate: true
+          draft: true
+          bodyFile: ${{ needs.fetch_updated_release_notes_file.outputs.updated_release_notes_filepath }}
```

Comment:
I would move this to the `env` list, like it is done for the `VERSION_NAME` and use it here as `${{ env.RELEASE_NOTES_FILE }}`